src/libs/logging.ts

```diff
@@ -0,0 +1,243 @@
+/* eslint-disable vue/max-len */
+/* eslint-disable prettier/prettier */
+/* eslint-disable max-len */
+import { useStorage } from '@vueuse/core'
+import { differenceInMilliseconds, differenceInSeconds, format } from 'date-fns'
+import localforage from 'localforage'
+import Swal from 'sweetalert2'
+
+import { useMainVehicleStore } from '@/stores/mainVehicle'
+
+import { degrees } from './utils'
+
+/**
+ * Variables data can be datalogged
+ */
+export enum DatalogVariable {
```

Comment:
Cant we use registerUsage to do this kind of definition for us dynamically ?

Reply:
If we do that, we lose the ability to add to the logs variables that are not being used by widgets.

Reply:
> Logs should be independent from video subtitles.

They are. It's the other way around. The subtitles are dependent on the logs.

> So.. why not having the log being done by the abstract vehicle class over registerUsage being used over widgets if the idea is to log everything ?
The registerUsage approach is just to deliver a good default to users. If they never configured which variables to show in the subtitle overlays, the default ones will be the ones they are using.